js/src/traceable.ts

```diff
@@ -351,7 +355,11 @@ const getTracingRunTree = <Args extends unknown[]>(
       | undefined
   );
   runTree.attachments = attached;
-  runTree.inputs = handleRunInputs<Args>(args, processInputs);
+  const processedInputs = handleRunInputs<Args>(args, processInputs);
+  if (isAsyncFn(processInputs)) {
+    runTree._awaitInputsOnPost = true;
```

Comment:
@dqbd sadly the new AI SDK 6 `output.responseFormat` which is a serialized representation of a given output is a Promise, and getting the run tree here must be synchronous. To get around this we can just await the promise when `runTree.postRun()` gets called